static/js/utils.ts

```diff
@@ -264,3 +265,37 @@ export function adjustInputValue(value: number, schema: Record<string, unknown>)
 
   return value;
 }
+
+type WoTOperation = keyof typeof Constants.WoTOperation;
+type WoTFormOperation = WoTOperation | WoTOperation[] | undefined;
+/**
+ * Finds the gateway api endpoint for a specific operation.
+ * It uses the assumption that gateway enpoints are pushed to
+ * the form array as the last element.
+ *
+ * @param {Array} forms The list of forms.
+ * @param {string} operation
+ * @returns {string | undefined} the href of the select form or undefined if not found.
+ */
+export function selectFormHref(
+  forms: Array<{ href: string; op: WoTFormOperation }>,
+  operation: WoTOperation
+): string | undefined {
+  return [...forms].reverse().find((selectedForm) => {
+    try {
+      const { protocol } = new URL(selectedForm.href);
+      return (
+        protocol === 'http:' &&
+        (!selectedForm.op || selectedForm.op === operation || selectedForm.op?.includes(operation))
+      );
+    } catch (error) {
+      if (error instanceof TypeError) {
+        // URL is relative or not well formatted
```

Comment:
I think I'm right in saying that all the form `href`s added by the gateway are just paths rather than absolute URLs (e.g. `/things/foo/properties/on`). That means this logic would always be triggered and the protocol check would never be applied.

Rather than just ignore the protocol if not an absolute URL, I think if you're going to check the protocol what you need to do here is resolve the `href` against the `base` member of the Thing Description first.

This is probably strictly not necessary since the last form entry (added by the gateway) will always be HTTP, so we could just remove the protocol check if you prefer (see below).